subsys/bluetooth/services/hids_c.c

```diff
@@ -941,25 +941,57 @@ int bt_gatt_hids_c_rep_write(struct bt_gatt_hids_c *hids_c,
 	return 0;
 }
 
+/**
+ *  @brief Process report write without response
+ *
+ *  @param conn Connection object.
+ *  @param rep_ptr Report object.
+ */
+static void rep_write_wo_rsp_process(struct bt_conn *conn, void *rep_ptr)
+{
+	struct bt_gatt_hids_c_rep_info *rep = rep_ptr;
+
+	if (!rep->write_cb) {
+		LOG_ERR("No write callback present");
+		return;
+	}
+	rep->write_cb(rep->hids_c, rep, 0);
+	rep->write_cb = NULL;
+}
+
 int bt_gatt_hids_c_rep_write_wo_rsp(struct bt_gatt_hids_c *hids_c,
 				   struct bt_gatt_hids_c_rep_info *rep,
+				   bt_gatt_hids_c_write_cb func,
```

Comment:
Please place this argument as the last argument. Similar to `bt_gatt_write_without_response_cb`